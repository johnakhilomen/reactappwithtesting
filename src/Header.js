import React, { Component } from 'react';

class Header extends Component {
    state = {  }
    render() { 
        return ( 
            <div>
                <center>
                    <h1>Header</h1>
                </center>
            </div>
         );
    }
}
 
export default Header;