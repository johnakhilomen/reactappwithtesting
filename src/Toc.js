import React, { Component } from 'react';

class Toc extends Component {
    state = {  }
    render() { 
        return ( 
            <div>Toc section</div>
         );
    }
}
 
export default Toc;