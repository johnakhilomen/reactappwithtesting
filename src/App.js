import React, {Component} from 'react';
import logo from './logo.svg';
import './App.css';
import Header from './Header';
import Toc from "./Toc";

class App extends Component {
  state = { 
    pgTitle: "A React App With Testing",
    menu: [
      {"Home" : "/home"},
      {"Register" : "/register"},
      {"Login" : "/login"},
      {"About us" : "/about us"}
    ]
   }
  render() { 
    return ( 
      <div>
        <Header></Header>
        <p>This is App</p>
        <Toc></Toc>
      </div>
     );
  }
}
 
export default App;

