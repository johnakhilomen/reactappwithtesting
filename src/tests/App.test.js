import React from 'react';
import App from '../App';
import Header from "../Header";
import Toc from "../Toc";
import renderer from 'react-test-renderer';
import {shallow} from "enzyme";
import ConfigureEnzymeAdapter from "./ConfigureEnzymeAdapter";
import {createAppWrapper} from "./TestWrappers";

//Configure Enzyme Adapter
ConfigureEnzymeAdapter.init();
const stringDotEmpty = "";
/**
 * @function createAppComponent
 */
const createAppComponent = () => {
  return renderer.create(<App></App>);
}
const appComponentTree = JSON.parse(JSON.stringify(createAppComponent()) );
//console.log(appComponentTree);

test("test that App renders without errors", ()=>{
  expect(appComponentTree).toMatchSnapshot();
})

appComponentTree.children[0].children.forEach(element => {
  //console.log(element);
});
test("test that App component renders main Div", ()=> {
/*let foundDiv = stringDotEmpty
Object.entries(appComponentTree).forEach(([k ,v])=> {
    if(k == "type" && v == "div")
    {
      foundDiv = "found";
      return;
    }
   });*/
  expect(appComponentTree.type).toBe("div");
})

test("test that App component doesn't render main Div", ()=> {
    expect(appComponentTree.type).not.toBe("");
  })


test("test that App component renders Header component", ()=> {
  const headerComponent =  createAppWrapper().find(Header);
  expect(headerComponent.length).toBe(1);
})

test("test that App component doesn't render Header component", ()=> {
  const headerComponent =  createAppWrapper().find(Header);
  expect(headerComponent.length).not.toBe(0);
})

test("test that App component renders Toc", ()=> {
  const TocComponent = createAppWrapper().find(Toc);
  expect(TocComponent.length).toBe(1);
})

test("test that App component doesn't render Toc", ()=> {
  const TocComponent = createAppWrapper().find(Toc);
  expect(TocComponent.length).not.toBe(0);
})

/*test("set headerText state when button is clicked", ()=> {
  const button = appWrapper.find("[className='myButton']");
  button.simulate("click");
  const changeHeaderText =  appWrapper.find(Header);
  expect(changeHeaderText.props().headerText).toBe("New Header Text");
})*/
