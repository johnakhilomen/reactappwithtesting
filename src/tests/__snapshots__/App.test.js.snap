// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that App renders without errors 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                "Header",
              ],
              "props": Object {},
              "type": "h1",
            },
          ],
          "props": Object {},
          "type": "center",
        },
      ],
      "props": Object {},
      "type": "div",
    },
    Object {
      "children": Array [
        "This is App",
      ],
      "props": Object {},
      "type": "p",
    },
    Object {
      "children": Array [
        "Toc section",
      ],
      "props": Object {},
      "type": "div",
    },
  ],
  "props": Object {},
  "type": "div",
}
`;
